content/articles/spring-boot-recaptcha-v3/index.md

```diff
@@ -0,0 +1,855 @@
+- [Create new site with Google to verify your captcha response]()
+- [Create a SpringBoot application using spring initializr]()
+- [Configure database connection properties]()
+- [Create User, Role, and ReCaptchaResponse models]()
+- [Create User and Role repositories]()
+- [Add a `CommandLineRunner` to create custom users]()
+- [Create a user service to handle user details]()
+- [Add `WebSecurityConfigurerAdapter` configuration class to handle login ]()
+- [Create ReCAPTCHA handler to get the score]()
+- [Create a Login Filter to redirect the user based on the score]()
+- [Add the `addFilterBefore()` to `HttpSecurity`]()
+- [Create a login and OTP Controller]()
+- [Create a login and OTP form and a default success page]()
+- [Test the login when the score is greater than 0.5]()
+- [Test the login when the score is less than 0.5]()
+
+### Prerequisites
+- Spring Boot
+- Spring Data JPA
+- Spring Security
+- Thymeleaf
+- JDK 11+
+- Intellij IDE
+
+### Create a new site with Google to verify your captcha response
+[Create a new reCAPTCHA V3 site ](https://www.google.com/recaptcha/admin/create) with any name you want, but the domain can be localhost as we will be testing the application locally. Take note of the site key and secret key, which we will use later in the application.
+
+### Create a SpringBoot application using spring initializr
+Go to [Spring Initialzr](https://start.spring.io/) and generate a new project with dependency `Spring Web`, `Spring Security`, `Spring Data JPA`, `Thymeleaf`, `Spring Boot Dev Tools`, `MySQL Driver` and `Lombok`.
+
+Import the generated project into Intellij and ensure you have an active internet connection to download the dependencies from the remote central repository.
+
+![Spring Initializr image](spring-initializr.png)
+
+### Configure database connection properties
+Since we will not deploy the application to production, the database tables, and data will be automatically created for us during application initialization using `spring.jpa.hibernate.ddl-auto=create` property.
+
+The SQL statements that will be generated will be displayed to the console to see what is happening behind the scenes and this will be enabled by `spring.jpa.show-sql=true` property.
+
+To ensure that we read the SQL statements easily, we can ensure they are formatted well by including the `spring.jpa.properties.hibernate.format_sql=true` property.
+
+application.properties
+```properties
+spring.datasource.url=jdbc:mysql://localhost:3306/recaptcha
+spring.datasource.username=username
+spring.datasource.password=password
+spring.jpa.hibernate.ddl-auto=create
+spring.jpa.show-sql=true
+spring.jpa.properties.hibernate.format_sql=true
+spring.jpa.properties.hibernate.dialect=org.hibernate.dialect.MySQL8Dialect
+```
+
+### Create User, Role, and ReCaptchaResponse models
+We will make use of project `Lombok` to generate getters and setters, no-argument constructor, and all arguments constructor for all the models that will be used.
```

Comment:
Change to "We will use project `Lombok` to generate getters and setters, no-argument constructor, and all arguments constructor for all the models in the project."